pytorch_lightning/plugins/ddp_sequential_plugin.py

```diff
@@ -0,0 +1,403 @@
+
+    @property
+    def data_parallel_group(self):
+        return mpu.get_data_parallel_group()
+
+    @property
+    def is_main_rpc_process(self) -> bool:
+        return self.main_rpc_process
+
+    @property
+    def return_after_exit_rpc_process(self) -> bool:
+        return True
+
+    def barrier(self, name: Optional[str] = None) -> None:
+        if torch_distrib.is_initialized() and self.is_main_rpc_process:
+            torch_distrib.barrier(group=self.data_parallel_group)
+
+    def _check_pipe_available(self):
+        if not FAIRSCALE_PIPE_AVAILABLE:
+            raise MisconfigurationException(
+                'PipeRPCPlugin requires FairScale and currently is only supported on PyTorch 1.6.'
+            )
+
+
+class LightningPipeModule(nn.Module):
+    """
+        This class wraps Fairscale Pipe and PipeRCPWrapper class.
+    """
+
+    def __init__(
+            self,
+            module: nn.Sequential,
+            balance: List[int],
+            microbatches: int = 8,
+            checkpoint='never'):
+        super().__init__()
+        self.module = module
+        self.balance = balance
+        self.microbatches = microbatches
+        self.checkpoint = checkpoint
+        self._init_pipe()
+
+    def _init_pipe(self):
+        device = torch.device("cuda", torch_distrib.get_rank())
+
+        self.module = PipeRPCWrapper(
+            module=self.module,
+            balance=self.balance,
+            chunks=self.microbatches,
+            style=PipelineStyle.MultiProcess,
+            input_device=device,
+            worker_map=self.get_worker_map(),
+            checkpoint=self.checkpoint,
+        )
+
+    def foreach_worker(self, *args, **kwargs):
+        self.module.foreach_worker(*args, **kwargs)
+
+    def forward(self, *args, **kwargs):
+        x = self.module(*args, **kwargs)
+        return x
+
+    def get_worker_map(self):
+        # TODO, is this correct with multinodes? We also assume "worker" is the same as defined in the RPCPlugin
+        return {rank: f"worker{rank}" for rank in range(torch_distrib.get_world_size())}
+
+
+def register_optimizers(ctx, model):
+    optimizers, lr_schedulers, optimizer_frequencies = model.trainer.init_optimizers(model)
+    model.trainer.optimizers = optimizers
+    model.trainer.lr_schedulers = lr_schedulers
+    model.trainer.optimizer_frequencies = optimizer_frequencies
+    model.trainer.convert_to_lightning_optimizers()
+
+
+def run_optimizer(ctx, model):
+    trainer = model.trainer
+    opt_idx = ctx["opt_idx"]
+    optimizer = trainer.optimizers[opt_idx]
+    optimizer.step(*ctx["args"], **ctx["kwargs"])
+
+
+def save_layers_on_all_rank_zero_workers(ctx, model):
+    gpus_per_model = ctx["gpus_per_model"]
+    rank = torch_distrib.get_rank()
+    if rank in range(gpus_per_model):
+        seq = list(model.children())[0]
+        torch.save(seq, f"seq_{rank}.pt")
+
+
+def reload_sequential_from_saved_layers(gpus_per_model):
```

Comment:
```suggestion
def load_sequential_from_saved_layers(gpus_per_model):
```
the "re" is confusing as you delete the files, so if you call it twice it fails for missing files, right?